onegittorulethemall/services/pagure.py

```diff
@@ -174,22 +175,26 @@ def fork_create(self):
     @property
     def fork(self):
         """PagureRepo instance of the fork of this repo."""
-        kwargs = replace_username_with_username(self.pagure_kwargs)
-        kwargs.update(
-            repo=self.repo,
-            namespace=self.namespace,
-            username=self.username,
-            instance_url=self.instance_url,
-            token=self.token,
-            is_fork=True,
-        )
-        fork_project = PagureProject(**kwargs)
-        try:
-            if fork_project.exists and fork_project.pagure.parent:
-                return fork_project
-        except:
-            return None
-        return None
+        if self.forked_project is None:
+            kwargs = sanitize_fork_username(self.pagure_kwargs)
+            kwargs.update(
+                repo=self.repo,
+                namespace=self.namespace,
+                username=self.username,
+                instance_url=self.instance_url,
+                token=self.token,
+                is_fork=True,
+            )
+            self.forked_project = PagureProject(**kwargs)
+            try:
+                # why do we do this harakiri?
+                if not (self.forked_project.exists and
+                        self.forked_project.pagure.parent):
+                    self.forked_project = None
+            except Exception as ex:
+                logger.info("exception while getting the forked project: %s", ex)
+                self.forked_project = None
+        return self.forked_project
```

Comment:
We can go with this but other work on this API should be done [here](https://github.com/user-cont/ogr).

Reply:
we can merge and I'll open a PR at ogr then with the changes